psi4/driver/p4util/python_helpers.py

```diff
@@ -661,18 +676,20 @@ def _qcvar_reshape_get(key, val):
     """For QCVariables where the 2D psi4.core.Matrix shape is unnatural, convert to natural shape in ndarray."""
 
     reshaper = None
-    if key.upper() == "MBIS CHARGES":
-        return val.np
-    elif key.upper() == "MBIS DIPOLES":
-        reshaper = (-1, 3)
-    elif key.upper() == "MBIS QUADRUPOLES":
-        val = val.np.reshape(-1, 6)
-        val = np.array([_multipole_plumper(val[iat], 2) for iat in range(len(val))])
-        return val
-    elif key.upper() == "MBIS OCTUPOLES":
-        val = val.np.reshape(-1, 10)
-        val = np.array([_multipole_plumper(val[iat], 3) for iat in range(len(val))])
-        return val
+    if key.upper().startswith("MBIS"):
+        if key.upper().endswith("CHARGES"):
+            return val.np
+        elif key.upper().endswith("DIPOLES"):
+            reshaper = (-1, 3)
+            return val.np.reshape(reshaper)
+        elif key.upper().endswith("QUADRUPOLES"):
+            val = val.np.reshape(-1, 6)
+            val = np.array([_multipole_plumper(val[iat], 2) for iat in range(len(val))])
+            return val
+        elif key.upper().endswith("OCTUPOLES"):
+            val = val.np.reshape(-1, 10)
+            val = np.array([_multipole_plumper(val[iat], 3) for iat in range(len(val))])
+            return val
```

Comment:
is mirroring the other structure by separating into two-level if-mbis/if-n-pole the only change here?

Reply:
yep the logic is unchanged here, just reorganized to be a bit more maintainable